tests/nosetests/pyanaconda_tests/module_source_url_test.py

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Jiri Konecny <jkonecny@redhat.com>
+#
+import unittest
+
+from tests.nosetests.pyanaconda_tests import check_dbus_property
+from pyanaconda.modules.common.constants.interfaces import PAYLOAD_SOURCE_URL
+from pyanaconda.modules.common.errors import InvalidValueError
+from pyanaconda.modules.common.structures.payload import URLData
+from pyanaconda.modules.payloads.constants import SourceType, URLSourceType
+from pyanaconda.modules.payloads.source.url.url import URLSourceModule
+from pyanaconda.modules.payloads.source.url.url_interface import URLSourceInterface
+
+
+class URLSourceInterfaceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.url_source_module = URLSourceModule()
+        self.url_source_interface = URLSourceInterface(self.url_source_module)
+
+    def _check_dbus_property(self, *args, **kwargs):
+        check_dbus_property(
+            self,
+            PAYLOAD_SOURCE_URL,
+            self.url_source_interface,
+            *args, **kwargs
+        )
+
+    def type_test(self):
+        """Test URL source has a correct type specified."""
+        self.assertEqual(SourceType.URL.value, self.url_source_interface.Type)
+
+    def set_url_base_source_properties_test(self):
+        url = URLData()
+        url.url = "http://example.com/repo"
+        url.url_type = URLSourceType.BASE.value
+
+        self._check_dbus_property(
+            "URL",
+            URLData.to_structure(url)
```

Comment:
It is good to have one test with the raw representation of the data structures to make sure that there are no surprises.

Reply:
Fixed.